### ISSUE TITLE
Bug: flow doesn't understand setting nullable properties

### ISSUE THREAD

**Issue by jaredly** [facebook/flow#603]
```js
type Obj = {
  doot: ?() => void
};

var obj: Obj = {doot: null};

obj.doot = function() {}
// this triggers two flow errors
// ----
// error| call of method doot
// || Function cannot be called on possibly null value
// error| null
// ---
// error| call of method doot
// || Function cannot be called on possibly undefined value
// error| undefined
obj.doot();
```

Using a top-level variable, on the other hand, works fine:

```js
var thing: ?() => void = null;

thing = function() {}
// this does not trigger a flow error
thing();
```

It also fails in the es6 class case (which is where I noticed the problem). See all the code on tryflow:

http://tryflow.org/?ex=2#97e488c2ac95b8ec04d59bcc17b53e99

**Comment by gabelevi** [comment]
@bhosmer fixed this with https://github.com/facebook/flow/commit/ef8ece919a4dbfde08cbb2c68c0335df2535aa8d and it will be out in the next release!

**Comment by samwgoldman** [comment]
Fixed in 0.14
